examples/quickstart-lerobot/lerobot_example/client_app.py

```diff
@@ -0,0 +1,80 @@
+"""huggingface_example: A Flower / Hugging Face LeRobot app."""
+
+import warnings
+
+import torch
+from flwr.client import Client, ClientApp, NumPyClient
+from flwr.common import Context
+from transformers import logging
+from logging import INFO
+
+from flwr.common.logger import log
+from lerobot_example.task import (
+    train,
+    test,
+    load_data,
+    set_params,
+    get_params,
+    get_model,
+)
+
+warnings.filterwarnings("ignore", category=FutureWarning)
+
+# To mute warnings reminding that we need to train the model to a downstream task
+# This is something this example does.
+logging.set_verbosity_error()
+
+
+# Check if GPU is available
+if torch.cuda.is_available():
+    nn_device = torch.device("cuda")
+    print("GPU is available. Device set to:", nn_device)
+else:
+    nn_device = torch.device("cpu")
+    print(f"GPU is not available. Device set to: {nn_device}. Inference will be slower than on GPU.")
```

Comment:
For simulations, it's preferable if the device is "auto discovered" at run time. This is because Flower's simulation engine can mask out devices at run time (e.g. tell `ClientApp`s to run on CPU-only even if a GPU is available). See my comment later in `client_fn()`.

Reply:
Makes sense. I did not realize how Flower manages hardware resource availability for clients at runtime.